dvc/ignore.py

```diff
@@ -72,36 +60,41 @@ def __call__(self, root, dirs, files):
 
 
 class DvcIgnoreFilter(object):
-    def __init__(self, wdir, ignore_file_handler=None):
+    def __init__(self, wdir):
         self.ignores = [
             DvcIgnoreDir(".git"),
             DvcIgnoreDir(".hg"),
             DvcIgnoreDir(".dvc"),
             DvcIgnoreFile(".dvcignore"),
         ]
 
-        self.ignore_file_handler = ignore_file_handler
+        from dvc.repo import Repo
+
+        try:
+            self.tree = WorkingTree(Repo.find_root())
```

Comment:
I don't think so. In case when we have .dvcignore in repository root, and have `data` folder, and run dvc_walk(`data`), we will not apply ignores contained in repository root to `data` content.

Reply:
But if we don't pass tree, no `.dvcignore` files would be applied, because `.update()` is skipped. Is this intended?

Reply:
And anyway if we don't pass `wdir` then it will use cwd, which could be anything. So tree is always made based on repo we are currently in.

Reply:
Ahh, sorry, I just realized that you were talking about passign `wdir` to `find_root`. I agree with that completely. Also, as Ruslan mentioned below, we cannot ignore external directory. I quess the best way to proceed will be to try find repo root, and if it cannot be found, use `wdir` as WorkingTree root.
Thought in such case we should mention in documentation that DVC will not handle `.dvcignore` files if they are outside of target external directory